tests/ga/test_collect_telemetry_events.py

```diff
@@ -527,4 +531,41 @@ def test_it_should_not_send_event_where_message_is_empty_and_report_event(self):
                 self._assert_handler_data_in_event_list(telemetry_events, extensions_with_count, expected_count=0)
                 self._assert_invalid_extension_error_event_reported(mock_event, extensions_with_count,
                                                                     InvalidExtensionEventError.EmptyMessageError,
-                                                                    expected_drop_count=1)
\ No newline at end of file
+                                                                    expected_drop_count=1)
+
+    def test_it_should_not_process_events_if_telemetry_service_stopped(self):
+        event_list = []
+        telemetry_handler = MagicMock(autospec=True)
+        telemetry_handler.stopped = MagicMock(return_value=True)
+        telemetry_handler.enqueue_event = MagicMock(wraps=event_list.append)
+
+        with self._create_extension_telemetry_processor(telemetry_handler) as extension_telemetry_processor:
+            self._create_random_extension_events_dir_with_events(3, self._WELL_FORMED_FILES)
+            extension_telemetry_processor.run()
+
+            self.assertEqual(0, len(event_list), "No events should have been enqueued")
+
+    def test_it_should_not_delete_event_files_except_current_one_if_service_stopped_midway(self):
+        event_list = []
+        telemetry_handler = MagicMock(autospec=True)
+        telemetry_handler.stopped = MagicMock(return_value=False)
+        telemetry_handler.enqueue_event = MagicMock(side_effect=ServiceStoppedError("Telemetry service stopped"),
+                                                    wraps=event_list.append)
+        no_of_extensions = 3
+        # self._WELL_FORMED_FILES has 3 event files, i.e. total files for 3 extensions = 3 * 3 = 9
+        # But since we delete the file that we were processing last, expected count = 8
```

Comment:
Why do we only delete the last file? Is this test supposed to mock the service getting stopped as we are reading the last file? I'm having trouble seeing it in this test.

Reply:
This is specifically by design. The idea is basically that if we're processing an event file and the service is stopped midway, we dont want to process that file again to avoid event duplication, so we delete it. (Note: In the extension telemetry events, 1 file can contain multiple events). The tradeoff is that we would loose the remaining events in that particular file but avoid duplication

Reply:
Ok, now I understand the intent of the test better, but I'm still having issues understanding the implementation. Where are you stopping the service midway?

I see this, but it was added at the beginning:
`telemetry_handler.enqueue_event = MagicMock(side_effect=ServiceStoppedError("Telemetry service stopped"),
                                                    wraps=event_list.append)`

Reply:
Basically in this test I'm raising a `ServiceStoppedError` whenever someone tries to enqueue any event using the `telemetry_handler.enqueue_event` function. Since this function is only called midway into the collect_events service, this is how I'm ensuring that its called in the middle :)